File: Linked-Lists/Java/Deleting_val_from_LL.java
import javax.swing.*;

class Node7{
    public
    int data;
    Node next;
    public
    Node7(int data1,Node next1)
    {
        data=data1;
        next=next1;
    }
    public
    Node7(int data1)
    {
        data=data1;
        next=null;
    }
};
public class Deleting_val_from_LL

{

    public static void print(Node  head)
    {
        Node temp=head;
        while(temp!=null)
        {
            System.out.print(temp.data+" ");	;
            temp=temp.next;
        }
    }
    public static  Node remove(Node head,int val)
    {
        if(head==null ) return null;
        if(head.data==val)
        {
            head=head.next;
            return head;
        }

        Node temp=head;
        Node prev=null;
        while(temp!=null){

            if(temp.data==val)
            {
                prev.next=prev.next.next;
                break;
            }
            prev=temp;
            temp=temp.next;
        }

        return head;

    }
    public static Node convertarraytolist(int[]  arr)
    {
        Node head=new Node(arr[0]);
        Node mover=head;
        for(int i=1;i<arr.length;i++)
        {
            Node temp=new Node(arr[i]);
            mover.next=temp;
            mover=temp;
        }
        return head;
    }

    public static void main(String[] args) {
        int[] arr={15,5,8,7,2,1};
        Node head=convertarraytolist(arr);
        head=remove(head,8);
        print(head);




    }
}

File: Linked-Lists/C++/deleting val from LL.cpp
#include <vector>
#include <iostream>
using namespace std;
class Node{
    public:
    int data;
    Node* next;
    public:
    Node(int data1,Node* next1)
    {
        data=data1;
        next=next1;
    }
    public:
    Node(int data1)
    {
        data=data1;
        next=nullptr;
    }
};
 Node* convertarraytolist(vector<int> &arr)
    {
    	Node* head=new Node(arr[0]);
    	Node* mover=head;
    	for(int i=1;i<arr.size();i++)
    	{
    		Node* temp=new Node(arr[i]);
    		mover->next=temp;
    		mover=temp;
		}
		return head;
	}
	
void print(Node * head)
{
Node* temp=head;
while(temp!=NULL)
{
cout<<temp->data<<" "	;
temp=temp->next;
}
}
Node* remove(Node *head,int val)
{
	
	if(head==NULL) return head;
	if(head->data==val)
	{
		Node * temp=head;
		head=head->next;
		free(temp);
		return head;
	}
	
	Node* temp=head;
	Node* prev=NULL;

	while(temp!=NULL){
	if(temp->data==val)
	{
		prev->next=prev->next->next;
		free(temp);
		break;
	}
		prev=temp;
		temp=temp->next;
	}
	return head;

	
}
int main() {
    // Write C++ code here
  
    vector<int>arr={15,5,8,7,1,6,2};
   Node * head=convertarraytolist(arr);
    head=remove(head,7);
    print(head);
    
    
    

    return 0;
}